app/services/immunization_service/send_sms_service.rb

```diff
@@ -0,0 +1,43 @@
+# app/services/immunization_service/send_sms_service.rb
+require 'net/http'
+require 'uri'
+require 'json'
+require 'sidekiq'
+
+module SendSmsService       
+    include Sidekiq::Worker
+
+    def self.perform_async(date, details) 
+      @date = Date.parse(date)
+      @details = details
+      @converted_date = @date.strftime('%d-%B-%Y')
+            
+      message = "Okondweda #{@details[:firstname]} #{@details[:sirname]},\n" \
+                "Mukukuziwitsidwa kuzabwelaso ku Chiptala\n" \
+                "pa tsiku **#{@converted_date}**.\n" \
+                "kut muzalandileso katemela.\n" \
+                "Zikomo."
+
+      uri = URI.parse("https://gateway.seven.io/api/sms")
```

Comment:
I am thinking API key and gateway should be in a configuration file. eg application.yml

Reply:
I have added all the suggested changes tried it and the job logs seems to OK